### PR TITLE
Feature/logging config

### PR DESCRIPTION
**Pull request by softwarejimenez**
:white_check_mark: I have updated the documentation accordingly.
:white_check_mark: I have read the CONTRIBUTING document.

### Summary

 In this example, we illustrate how to use the Logging APIs.

### CODE REVIEW

File: examples/connext_dds/logging_config/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+In this example we illustrate how to use the Logging API. As in other examples
+a DomainParticipant, Type, Topic, Publisher and DataWriter are created.
+
+The verbosity is set to **NDDS_CONFIG_LOG_VERBOSITY_WARNING**, it means that all
+warnings and erros will be filter in and will be logged in the terminal.
+
+The **NDDS_CONFIG_LOG_PRINT_FORMAT_MAXIMAL** is set in the *NDDS_Config_LogLevel*
+**NDDS_CONFIG_LOG_LEVEL_WARNING**, it means that all the warnings message will
+be composed wit all the available RTI Connext diagnostic fields.
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/README.md:28]:
```suggestion
be composed with all the available RTI Connext diagnostic fields.
```

File: examples/connext_dds/logging_config/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+Two samples are written and then for the second one we force the write operation
+to fail: when using *DDS_BY_SOURCE_TIMESTAMP_DESTINATIONORDER_QOS*, if source
+timestamp is older than the one from a previous write, a warning  will be logged.
+
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/README.md:33]:
```suggestion

The expected output is:
```

File: examples/connext_dds/logging_config/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+[0x01016FA9,0xB965C9C1,0x4E59B461:0x80000003{E=DW,T=Example logging,C=logging,D=0}|WRITE]
+```
+
+- **Module**: Id representing the module
+
+```plaintext
+Mx16
+```
+
+- **File and Line**
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/README.md:109]:
```suggestion
- **File and Line**: The file name (including the path to the file) and line number where to log message was printed.
```

File: examples/connext_dds/logging_config/README.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+```plaintext
+writer_history.1.0/srcC/memory/Memory.c:7411
+```
+
+- **Method** name where the message is logged
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/README.md:115]:
```suggestion
- **Method**: The method name where the message was logged
```

File: examples/connext_dds/logging_config/README.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+```plaintext
+WriterHistoryMemoryPlugin_addSample:RTI0x2161002:out of order
+```
+
+- **Message Id**: It is a identification of the message.
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/README.md:121]:
```suggestion
- **Message Id**: The unique identifier of the logged message.
```

File: examples/connext_dds/logging_config/c++/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# (c) 2020 Copyright, Real-Time Innovations, Inc. All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/c++/CMakeLists.txt:1]:
```suggestion
# (c) 2021 Copyright, Real-Time Innovations, Inc. All rights reserved.
```

File: examples/connext_dds/logging_config/c++/USER_QOS_PROFILES.xml [hunk 7]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0"?>
+<!--
+Description
+XML QoS Profile for logging
+
+(c) Copyright, Real-Time Innovations, 2020.  All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/c++/USER_QOS_PROFILES.xml:6]:
```suggestion
(c) Copyright, Real-Time Innovations, 2021.  All rights reserved.
```

File: examples/connext_dds/logging_config/c++/USER_QOS_PROFILES.xml [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            </datareader_qos>
+            <participant_qos>
+                <!--
+                The participant name, if it is set, will be displayed in the
+                RTI tools, making it easier for you to tell one
+                application from another when you're debugging.
+                -->
+                <participant_name>
+                    <name>loggingParticipant</name>
+                    <role_name>loggingParticipantRole</role_name>
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/c++/USER_QOS_PROFILES.xml:92]: Is there a specific reason we are giving the participant a role name?

File: examples/connext_dds/logging_config/c++/application.h [hunk 9]
```diff
@@ -0,0 +1,106 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2020.  All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/c++/application.h:2]:
```suggestion
* (c) Copyright, Real-Time Innovations, 2021.  All rights reserved.
```

Comment (samuelraeburn) [reply examples/connext_dds/logging_config/c++/application.h:2]: Edit: See above comment about this file

File: examples/connext_dds/logging_config/c++/application.h [hunk 10]
```diff
@@ -0,0 +1,106 @@
+/*
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/c++/application.h:1]: I think this file should be added to gitignore and not committed.

File: examples/connext_dds/logging_config/c++/logging.idl [hunk 11]
```diff
@@ -0,0 +1,15 @@
+/*
+ * (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/c++/logging.idl:2]:
```suggestion
 * (c) 2021 Copyright, Real-Time Innovations, Inc.  All rights reserved.
```

File: examples/connext_dds/logging_config/c++/logging_publisher.cxx [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+    // Create data for writing, allocating all members
+    logging *data = loggingTypeSupport::create_data();
+    if (data == NULL) {
+        return shutdown_participant(
+            participant,
+            "loggingTypeSupport::create_data error",
+            EXIT_FAILURE);
+    }
+
+    /* Write data */
```
Comment (samuelraeburn) [inline examples/connext_dds/logging_config/c++/logging_publisher.cxx:124]:
```suggestion
    // Write data
```

**Review by samuelraeburn — CHANGES_REQUESTED** [review]
I have reviewed the c++ example. Same feedback applies to all other language

File: examples/connext_dds/logging_config/c++/USER_QOS_PROFILES.xml (hunk 8 above)
Comment (softwarejimenez) [reply examples/connext_dds/logging_config/c++/USER_QOS_PROFILES.xml:92]: generated QoS. I have remove it.

File: examples/connext_dds/logging_config/c++/application.h (hunk 10 above)
Comment (softwarejimenez) [reply examples/connext_dds/logging_config/c++/application.h:1]: I would like to remove it too. But this file is needed if not the compilation will fail.

This file is not generated as part of `connextdds_add_example`.

**Review by samuelraeburn — APPROVED** [review]

**Pull request merged**